fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/security/PoPToken.java

```diff
@@ -8,4 +8,9 @@
   public PoPToken(byte[] privateKey, byte[] publicKey) {
     super(new PlainPrivateKey(privateKey), new PublicKey(publicKey));
   }
+
+  @Override
+  public PlainPrivateKey getPrivateKey() {
+    return (PlainPrivateKey) super.getPrivateKey();
+  }
```

Comment:
Is there a difference for PoPToken to return a PrivateKey or a PlainPrivateKey ? (is it only used for sign ?)
If it return a PlainPrivateKey, the private key can be accessed using the getData. Could this be a problem ?

Reply:
It is not the best, but it does not pose that much threat.
The private key might be needed for some math later or for storage.

It allows the wallet to be tested easily.

And most of all, removing it and the getter of KeyPair does not resolve the issue.
With this piece of code, if it is executed from the right scope, you can access any private field value as it is stored in memory :

```java
PoPToken pair = new PoPToken(VALID_PRIVATE_KEY, VALID_PUBLIC_KEY);
try {
  Field key = pair.getClass().getDeclaredField("privateKey");
  key.setAccessible(true);
  PlainPrivateKey privateKey = (PlainPrivateKey) key.get(pair);
  System.out.println(privateKey.getEncoded());
} catch (NoSuchFieldException | IllegalAccessException e) {
  e.printStackTrace();
}
```

If we really wanted to be secure, we should store the encoded key in memory. But this is way too advanced for the current PR